src/pretix/base/logentrytypes.py

```diff
@@ -0,0 +1,238 @@
+from django.utils.html import escape
+from django.utils.translation import gettext_lazy as _, pgettext_lazy
+
+from pretix.base.signals import EventPluginRegistry
+
+
+def make_link(a_map, wrapper, is_active=True, event=None, plugin_name=None):
+    if a_map:
+        if is_active:
+            a_map['val'] = '<a href="{href}">{val}</a>'.format_map(a_map)
+        elif event and plugin_name:
+            a_map['val'] = (
+                '<i>{val}</i> <a href="{plugin_href}">'
+                '<span data-toggle="tooltip" title="{errmes}" class="fa fa-warning fa-fw"></span></a>'
+            ).format_map({
+                **a_map,
+                "errmes": _("The relevant plugin is currently not active. To activate it, click here to go to the plugin settings."),
+                "plugin_href": reverse('control:event.settings.plugins', kwargs={
+                    'organizer': event.organizer.slug,
+                    'event': event.slug,
+                }) + '#plugin_' + plugin_name,
+            })
+        else:
+            a_map['val'] = '<i>{val}</i> <span data-toggle="tooltip" title="{errmes}" class="fa fa-warning fa-fw"></span>'.format_map({
+                **a_map,
+                "errmes": _("The relevant plugin is currently not active."),
+            })
+        return wrapper.format_map(a_map)
+
+
+class LogEntryTypeRegistry(EventPluginRegistry):
+    def new_from_dict(self, data):
+        """
+        Register multiple instance of a LogEntryType class with different action_type
+        and plain text strings, as given by the items of the specified data dictionary.
+
+        This method is designed to be used as a decorator as follows:
+
+        .. code-block:: python
+
+            @log_entry_types.new_from_dict({
+                'pretix.event.item.added': _('The product has been created.'),
+                'pretix.event.item.changed': _('The product has been changed.'),
+                # ...
+            })
+            class CoreItemLogEntryType(ItemLogEntryType):
+                # ...
+
+        :param data: action types and descriptions
+                     ``{"some_action_type": "Plain text description", ...}``
+        """
+        def reg(clz):
+            for action_type, plain in data.items():
+                self.register(clz(action_type=action_type, plain=plain))
+            return clz
+        return reg
+
+
+"""
+Registry for LogEntry types.
+
+Each entry in this registry should be an instance of a subclass of ``LogEntryType``.
+They are annotated with their ``action_type`` and the defining ``plugin``.
+"""
+log_entry_types = LogEntryTypeRegistry({'action_type': lambda o: getattr(o, 'action_type')})
+
+
+class LogEntryType:
+    """
+    Base class for a type of LogEntry, identified by its action_type.
+    """
+
+    def __init__(self, action_type=None, plain=None):
+        assert self.__module__ != LogEntryType.__module__  # must not instantiate base classes, only derived ones
```

Comment:
Good to know! I'd say it's safe enough to assume that if this assert fails, it will already be triggered during debugging or test execution. As LogEntryTypes are typically instanciated only once at application startup, it's unlikely this only fails in a specific situation on production. 

But it also wouldn't hurt to change it I guess?
```suggestion
        if self.__module__ == LogEntryType.__module__
            raise Exception("LogEntryType: must not instantiate base classes, only derived ones")
```